### PR TITLE
Support macOS local VMs in KMT

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR adds support for macOS VMs in the KMT framework:

* Use vmconfig templates to account for differences in OSs and paths
* Fix configuration paths for macOS 
* Reorganization of certain files in tasks/kernel_matrix_testing to avoid import loops
* Use libvirt socket path provided by OS configuration 
* Adapt some libvirt calls to macOS requirements
* Provide initialisation code for VMs in macOS 
* Update KMT Readme
* Fix kmt.ssh-config tasks for local VMs 
* 
### Motivation

https://datadoghq.atlassian.net/browse/EBPF-421

### Additional Notes

See additional developer documentation: https://datadoghq.atlassian.net/wiki/spaces/EBPFTEAM/pages/3478291163/Local+VM+support+for+KMT

User documentation:
https://github.com/DataDog/datadog-agent/blob/main/tasks/kernel_matrix_testing/README.md

### Describe how to test/QA your changes

Tested locally with different agent developers using macOS

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]
LGTM for files owned by Developer Tools and e2e testing teams

File: tasks/kernel_matrix_testing/stacks.py [hunk 1]
```diff
@@ -24,6 +26,28 @@
 ARM_INSTANCE_TYPE = "m6gd.metal"
 
 
+def _get_active_branch_name():
+    head_dir = Path(".") / ".git" / "HEAD"
+    with head_dir.open("r") as f:
```
Comment (chouetz) [inline tasks/kernel_matrix_testing/stacks.py:31]:
```suggestion
    with head_dir.open() as f:
```

File: tasks/kernel_matrix_testing/stacks.py [hunk 2]
```diff
@@ -24,6 +26,28 @@
...
+        content = f.read().splitlines()
+
+    for line in content:
+        if line[0:4] == "ref:":
```
Comment (chouetz) [inline tasks/kernel_matrix_testing/stacks.py:35]:
```suggestion
        if line.startswith("ref:"):
```

File: tasks/kernel_matrix_testing/download.py [hunk 3]
```diff
@@ -72,18 +57,20 @@ def download_rootfs(ctx, rootfs_dir, vmconfig_template):
...
             continue
 
         for disk in vmset["disks"]:
-            d = os.path.basename(disk["source"])
+            # Use the uncompressed disk name, avoid errors due to images being downloaded but not extracted
+            d = os.path.basename(disk["target"])
             if not os.path.exists(os.path.join(rootfs_dir, d)):
-                disks_to_download.append(d)
+                if d.endswith(".xz"):
+                    d = d[: -len(".xz")]
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kernel_matrix_testing/download.py:69]: Yes, the images are generated in the [ami-builder repository](https://github.com/DataDog/ami-builder/tree/master/scripts/kernel-version-testing) and we always compress with `xz`.

File: tasks/kernel_matrix_testing/env-setup.sh
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kernel_matrix_testing/env-setup.sh:1]: Changed back, I had a misconfigured autoformatter.

File: tasks/kernel_matrix_testing/kmt_os.py [hunk 4]
```diff
@@ -40,16 +54,116 @@ def restart_libvirtd(ctx, sudo):
...
 
     @staticmethod
     def assert_user_in_docker_group(_):
         return True
+
+    @staticmethod
+    def init_local(ctx: Context):
+        # Configure libvirt sockets
+        ctx.run(
+            f"gsed -i -E 's%(# *)?unix_sock_dir = .*%unix_sock_dir = \"{MacOS.libvirt_system_dir}\"%' {MacOS.libvirt_conf}"
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kernel_matrix_testing/kmt_os.py:92]: Added, thanks!

File: tasks/kernel_matrix_testing/libvirt.py [hunk 5]
```diff
@@ -1,5 +1,12 @@
+import sys
+
 from tasks.kernel_matrix_testing.tool import info
 
+try:
+    import libvirt
+except ImportError:
+    libvirt = None
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply tasks/kernel_matrix_testing/libvirt.py:8]: The behavior is not the same. Here `libvirt` is an optional import for both macOS and Linux, people who aren't using KMT do not need to install it, so we don't want to raise any error if it's not present. In another PR we're working on better error messages when KMT users do not have the KMT-specific dependencies.

**Review by chouetz — APPROVED** [review]

File: tasks/kernel_matrix_testing/libvirt.py [hunk 6]
```diff
@@ -23,7 +30,13 @@ def delete_domains(conn, stack):
         name = domain.name()
         if domain.isActive():
             domain.destroy()
-        domain.undefine()
+
+        undefine_flags = 0
+        if sys.platform == "darwin":
+            undefine_flags |= libvirt.VIR_DOMAIN_UNDEFINE_NVRAM
```
Comment (usamasaqib) [inline tasks/kernel_matrix_testing/libvirt.py:36]: Would this prevent libvirt from deleting the `nvram` file?

**Review by usamasaqib — APPROVED** [review]

**Pull request merged**